docs/Components-Header.md

```diff
@@ -61,34 +158,85 @@ The `onLeftPress` is an optional prop that is used to set the function to be cal
 <Header leftIcon="back" onLeftPress={() => navigation.goBack()} />
 ```
 
-### `onRightPress`
+### `rightIcon`
 
-The `onRightPress` is an optional prop that is used to set the function to be called when the right navigation button is pressed.
+The `rightIcon` is an optional prop that is used to set the icon for the right navigation button. Custom icons can be created by using the [`Icon` component](./Components-Icon.md#custom-icons). Once you create a custom icon, just pass the string name of the icon to the `rightIcon` prop.
 
 ```tsx
-<Header rightIcon="bullet" onRightPress={() => Alert.alert("pressed")} />
+<Header rightIcon="back" onRightPress={() => navigation.goBack()} />
 ```
 
-### `style`
+### `rightIconColor`
 
-The `style` prop is an optional prop that is used to set the style of the header container. This is a [`StyleProp<ViewStyle>`](https://reactnative.dev/docs/view-style-props) object.
+The `rightIconColor` is an optional prop that is used to set the tint color of the right navigation icon.
 
 ```tsx
 <Header
-  headerText="Header Title"
-  onLeftPress={() => navigation.goBack()}
-  style={{ backgroundColor: "purple" }}
+  rightIcon="back"
+  onRightPress={() => navigation.goBack()}
+  rightIconColor="white"
 />
 ```
 
-### `titleStyle`
+### `rightText`
 
-The `titleStyle` prop is an optional prop that is used to set the style of the header title. This is a [`StyleProp<TextStyle>`](https://reactnative.dev/docs/text-style-props) object.
+The `rightText` is an optional prop that is used to set the text for the right navigation button. Overrides the `rightIcon` prop.
+
+```tsx
+<Header rightText="Back" onRightPress={() => navigation.goBack()} />
+```
+
+### `rightTx`
+
+The `rightTx` is an optional prop that is used to lookup the translation for the right navigation button. Overrides the `rightIcon` and `rightText` prop`.
+
+```tsx
+<Header rightTx="header.back" onRightPress={() => navigation.goBack()} />
+```
+
+### `rightTxOptions`
+
+The `rightTxOptions` is an optional prop that is used to pass props to the translation lookup for the right navigation button. This is useful if you need to pass in dynamic values to the translation.
 
 ```tsx
 <Header
-  headerText="Header Title"
-  onLeftPress={() => navigation.goBack()}
-  titleStyle={{ color: "white" }}
+  rightTx="header.back"
+  rightTxOptions={{ name: "John" }}
+  onRightPress={() => navigation.goBack()}
+/>
+```
+
+### `RightActionComponent`
+
+The `RightActionComponent` is an optional `ReactElement` prop that is used to set a custom component for the right navigation button. Overrides the `rightIcon`, `rightText`,  and `rightTx` props.
+
+```tsx
+<Header
+  RightActionComponent={<Text>Back</Text>}
+  onRightPress={() => navigation.goBack()}
```

Comment:
One thing i forgot to mention. When passing in a Right/Left ActionComponent, it overrides the props listed above in addition to the on Right/Left Press props (since that wraps the text/icons only).